src/Microsoft.AspNetCore.Mvc.Razor/ViewComponentTagHelperDescriptorProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Mvc.Razor.Internal;
+using Microsoft.AspNetCore.Mvc.ViewComponents;
+using Microsoft.AspNetCore.Razor.Compilation.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor
+{
+    public class ViewComponentTagHelperDescriptorProvider
+    {
+        public static IEnumerable<TagHelperDescriptor> GetDescriptorsFromAssembly(string assemblyName)
```

Comment:
How is this going to be used? It's a bit hard to review this since I don't know what the design for it is.